# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: xpcom/threads/nsITimer.idl
# Commit: 1585a4d5b845
# Full Hash: 1585a4d5b84595ca315da11f6ec684c0ae5c7cd3
# Author: Byron Campen [:bwc] <docfaraday@gmail.com>
# Date: 2021-09-04 09:40:24
# Regressor Bug: 1726177
# File Overlap Count: 1
# Description:
#   Bug 1726177: Update IDL file to better reflect previous behavior, and cover this behavior in the test-case. r=xpcom-reviewers,nika
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D123813
# ==============================================================================

diff -r 44e20dc3d292 -r 1585a4d5b845 xpcom/threads/nsITimer.idl
--- a/xpcom/threads/nsITimer.idl	Fri Sep 03 21:27:15 2021 +0000
+++ b/xpcom/threads/nsITimer.idl	Fri Sep 03 21:59:00 2021 +0000
@@ -90,12 +90,14 @@
 
   /**
    * A TYPE_REPEATING_PRECISE_CAN_SKIP repeating timer aims to have constant
-   * period between firings.  The processing time for each timer callback
-   * should not influence the timer period.  However this timer type
-   * guarantees that it will not queue up new events to fire the callback
-   * until the previous callback event finishes firing.  If the callback
-   * takes a long time, then the next callback will be scheduled immediately
-   * afterward, but only once.  This is the only non-slack timer available.
+   * period between firings.  The processing time for each timer callback will
+   * not influence the timer period.  If the callback finishes after the next
+   * firing(s) should have happened (either because the callback took a long
+   * time, or the callback was called extremely late), that firing(s) is
+   * skipped, but the following sequence of firing times will not be altered.
+   * This timer type guarantees that it will not queue up new events to fire
+   * the callback until the previous callback event finishes firing.  This is
+   * the only non-slack timer available.
    */
   const short TYPE_REPEATING_PRECISE_CAN_SKIP = 3;
 
